build-tools/packages/build-tools/src/common/npmPackage.ts

```diff
@@ -472,3 +472,29 @@ export class Packages {
 		return !results.some((result) => result.error);
 	}
 }
+
+/**
+ * Reads the contents of package.json, applies a transform function to it, then writes the results back to the source
+ * file.
+ *
+ * @param packageDir - The path to the directory containing package.json.
+ * @param packageTransformer - A function that will be executed on the package.json contents before writing it
+ * back to the file.
+ *
+ * @remarks
+ *
+ * The package.json is always sorted using sort-package-json.
+ */
+export function updatePackageJsonFile(
+	packageDir: string,
+	packageTransformer: (json: PackageJson) => void,
+): void {
+	const packagePath = path.join(packageDir, "package.json");
+	const indentation = detectIndent(packagePath).indent || "\t";
+	const pkgJson: PackageJson = readJsonSync(packagePath);
+
+	// Transform the package.json if needed
+	packageTransformer?.(pkgJson);
```

Comment:
you made this required again, so:
```suggestion
	// Transform the package.json
	packageTransformer(pkgJson);
```